packages/yew-macro/src/derive_props/field.rs

```diff
@@ -320,7 +318,29 @@ fn is_path_segments_an_option(path_segments: impl Iterator<Item = String>) -> bo
 ///
 /// Users can define their own [`Option`] type and this will return true - this is unavoidable.
 fn is_path_an_option(path: &Path) -> bool {
-    is_path_segments_an_option(path.segments.iter().take(3).map(|ps| ps.ident.to_string()))
+    is_path_segments_an_option(path.segments.iter().map(|ps| &ps.ident))
+}
+
+fn is_path_segments_a_string<'a, T>(mut iter: impl Iterator<Item = &'a T>) -> bool
+where
+    T: 'a + ?Sized + PartialEq<str>,
+{
+    iter.next().map_or(false, |first| {
+        first == "String"
+            || (first == "std" || first == "alloc")
+                && iter.next().map_or(false, |x| x == "string")
+                && iter.next().map_or(false, |x| x == "String")
+    })
+}
+
+/// returns true when the [`Path`] seems like a [`String`] type.
+///
+/// This function considers the following paths as Strings:
+/// - std::string::String
+/// - alloc::string::String
+/// - String
+pub(crate) fn is_path_a_string(path: &Path) -> bool {
+    is_path_segments_a_string(path.segments.iter().map(|ps| &ps.ident))
```

Comment:
```suggestion
pub(crate) fn is_path_a_string(path: &Path) -> bool {
    use syn::parse_quote as q;

    p == q!(String)
        || p == q!(alloc::string::String)
        || p == q!(std::string::String)
        || p == q!(::alloc::string::String)
        || p == q!(::std::string::String)
```

I think is_path_segments_a_string is only used for testing purpose, so we can simply check the path here instead.

This should also help with adding more variants in the future should we need it.

Reply:
I decided to benchmark both variants, here's the code of the benchmarks:
```rs
#![feature(test)]
extern crate test;

use syn::parse_quote as q;
use syn::Path;
use test::Bencher;

fn is_path_segments_a_string<'a, T>(mut iter: impl Iterator<Item = &'a T>) -> bool
where
    T: 'a + ?Sized + PartialEq<str>,
{
    iter.next().map_or(false, |first| {
        first == "String"
            || (first == "std" || first == "alloc")
                && iter.next().map_or(false, |second| second == "string")
                && iter.next().map_or(false, |third| third == "String")
    })
}

fn is_path_a_string_1(path: &Path) -> bool {
    is_path_segments_a_string(path.segments.iter().map(|ps| &ps.ident))
}

fn is_path_a_string_2(path: &Path) -> bool {
    path == &q!(String)
        || path == &q!(alloc::string::String)
        || path == &q!(std::string::String)
        || path == &q!(::alloc::string::String)
        || path == &q!(::std::string::String)
}

#[bench]
fn bench_1(b: &mut Bencher) {
    b.iter(|| {
        is_path_a_string_1(&q!(String));
        is_path_a_string_1(&q!(::std::String));
        is_path_a_string_1(&q!(std::string::String));
        is_path_a_string_1(&q!(Vec));
    })
}

#[bench]
fn bench_2(b: &mut Bencher) {
    b.iter(|| {
        is_path_a_string_2(&q!(String));
        is_path_a_string_2(&q!(::std::String));
        is_path_a_string_2(&q!(std::string::String));
        is_path_a_string_2(&q!(Vec));
    })
}

```
here's the output of `cargo bench`:
```console
running 2 tests
test bench_1 ... bench:       4,079 ns/iter (+/- 137)
test bench_2 ... bench:      23,215 ns/iter (+/- 541)

```
The first approach is as readable as the second one, but is much faster